pkg/server/node.go

```diff
@@ -431,34 +439,42 @@ func (s *nodeServer) getFetchBaseSVIDResponse(
 	return &node.FetchBaseSVIDResponse{SvidUpdate: svidUpdate}, nil
 }
 
-func (s *nodeServer) getSpiffeIDFromCtx(ctx context.Context) (spiffeID string, err error) {
+func (s *nodeServer) getCertFromCtx(ctx context.Context) (certificate *x509.Certificate, err error) {
 
 	ctxPeer, ok := peer.FromContext(ctx)
 	if !ok {
-		return "", errors.New("An SVID is required for this request")
+		return nil, errors.New("It was not posible to extract peer from request")
 	}
 	tlsInfo, ok := ctxPeer.AuthInfo.(credentials.TLSInfo)
-	if ok {
-		spiffeID, err := uri.GetURINamesFromCertificate(tlsInfo.State.PeerCertificates[0])
-		if err != nil {
-			return "", err
-		}
-		return spiffeID[0], nil
+	if !ok {
+		return nil, errors.New("It was not posible to extract AuthInfo from request")
 	}
-	return "", errors.New("It was not posible to read a SVID from your request")
+
+	if len(tlsInfo.State.PeerCertificates) == 0 {
+		return nil, errors.New("PeerCertificates was empty")
+	}
+
+	return tlsInfo.State.PeerCertificates[0], nil
 }
 
 func (s *nodeServer) signCSRs(
-	csrs [][]byte, regEntries []*common.RegistrationEntry) (
+	peerCert *x509.Certificate, csrs [][]byte, regEntries []*common.RegistrationEntry) (
 	svids map[string]*node.Svid, err error) {
 
+	uriNames, err := uri.GetURINamesFromCertificate(peerCert)
+	if err != nil {
+		s.l.Error(err)
+		return nil, errors.New("An SPIFFE ID is required for this request")
+	}
+	baseSpiffeID := uriNames[0]
```

Comment:
It is possible to call this endpoint with an SVID which is not a base SVID... so maybe naming this var `callerID` or something is better.

Similarly, we only want to perform the check for base SPIFFE IDs... so we should probably gate the logic below with something which checks for the base ID pattern, i.e. trustdomain/spire/agent/*

Reply:
:thumbsup: